contracts/base/UpgradeableProxy.sol

```diff
@@ -0,0 +1,14 @@
+import "@openzeppelin/upgrades/contracts/upgradeability/AdminUpgradeabilityProxy.sol";
+
+contract UpgradeableProxy is AdminUpgradeabilityProxy {
+    modifier ifAdmin() {
```

Comment:
Probably we could use ifAdmin() from BaseAdminUpgradeabilityProxy unless we need to throw 'UPGRADABLE_CALLER_MUST_BE_ADMIN' as a message.

Reply:
We want our own to have the `require` statement instead of `if` so that if an admin is required for a function it will revert rather than call the fallback. Which is the intended functionality we are looking for here.